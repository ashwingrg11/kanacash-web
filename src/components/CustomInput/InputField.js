import React from "react";
import { Input } from "./StyledComponent";
import PropTypes from "prop-types";
InputField.defaultProps = {
  customWidth: "20rem",
};
InputField.propTypes = {
  type: PropTypes.string,
  customWidth: PropTypes.string,
  onChangeValue: PropTypes.func,
  placeHolder: PropTypes.string,
  customName: PropTypes.string,
  customValue: PropTypes.any,
  style: PropTypes.any,
};
export default function InputField({
  type,
  customWidth,
  onChangeValue,
  placeHolder,
  customName,
  customValue,
  style,
}) {
  const onValueChange = (e) => {
    e.persist();
    let returnValue;
    if (e.target.value === "") {
      returnValue = {
        name: e.target.name,
        value: "",
      };
    } else {
      returnValue = {
        name: e.target.name,
        value: parseInt(e.target.value),
      };
    }
    onChangeValue(returnValue);
  };
  //   const [value, setValue] = React.useState("");
  //   React.useEffect(() => {
  //     if (customValue !== "") {
  // setValue()    }
  // }, [customValue]);
  return (
    <Input
      type={type}
      onChange={onValueChange}
      width={customWidth}
      placeholder={placeHolder}
      value={customValue}
      name={customName}
      style={style}
    />
  );
}
