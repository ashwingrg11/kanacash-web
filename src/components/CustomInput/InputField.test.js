import React from 'react';
import InputField from './InputField';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe('<InputField />', () => {
  it('InputField renders correctly', () => {
    const tree = renderer
      .create(
          <InputField />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});