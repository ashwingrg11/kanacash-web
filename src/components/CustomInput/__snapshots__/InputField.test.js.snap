// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> InputField renders correctly 1`] = `
<input
  className="sc-AxjAm gTCzCw"
  onChange={[Function]}
  width="20rem"
/>
`;
